lib/services/api/session_manager.ts

```diff
@@ -276,15 +280,51 @@ export class SNSessionManager extends PureService {
     });
   }
 
+  public async refreshSession() {
+    const userUuid = this.getUser()?.uuid;
+    const response = await this.apiService!.refreshSession(userUuid);
+
+    if (response.error) {
+      return {
+        response: this.apiService!.createErrorResponse(
+          messages.API_MESSAGE_GENERIC_INVALID_LOGIN
+        ),
+      } as SessionManagerResponse;
+    }
+
+    await this.handleRefreshSessionResponse(response);
+
+    return {
+      response
+    } as SessionManagerResponse;
+  }
+
   private async handleAuthResponse(response: HttpResponse) {
     if (response.error) {
       return;
     }
     const user = response.user;
     this.user = user;
     await this.storageService!.setValue(StorageKey.User, user);
-    const session = new Session(response.token);
+    const session = this.initializeSessionFromResponse(response);
     await this.storageService!.setValue(StorageKey.Session, session);
     await this.setSession(session);
   }
+
+  private async handleRefreshSessionResponse(response: HttpResponse) {
```

Comment:
This function should probably just be inline inside of refreshSession, if it's not reusable.